### DISCUSSION TITLE
verify combinatorial logic

### DISCUSSION THREAD

**Discussion by jreinauld** [cocotb/cocotb#3024] [Q&A]
Hello,

I am wondering how I can verify combinatorial logic with cocotb.
Say I have a combinatorial addr: 

```
signal siga : unsigned(7 downto 0);
signal sigb : unsigned(7 downto 0);
signal sigc : unsigned(7 downto 0);
sigc <= siga + sigb;
```

I can set `siga `and `sigb `easily from my python test.

But how can I check that `sigc` updates immediately (combinatorially), without waiting for any clock or whatsoever?

Thanks,

// Julien

**Comment by marlonjames** [comment]
You could use `await Edge(dut.sigc)` which will wait until any change is seen on `sigc`.

**Reply by jreinauld** [reply]
That is a good idea!
However in some case some input changes might not generate any output change.
(e.g. I test a and-reduce of 8 bits, 2 of these are already 0 and I set a third one to 0 as well, output was 0 already and remains 0)
So I think it does not work :(
